tests/test_console.py

```diff
@@ -58,43 +37,83 @@ def runner():
 
 def test_prints_an_empty_grid_correctly(empty_board_output, runner):
     empty_board_state = ['-', '-', '-', '-', '-', '-', '-', '-', '-']
+    game = MockGame(empty_board_state)
 
-    assert runner.render_board(empty_board_state) == empty_board_output
+    assert runner.render_board(game) == empty_board_output
 
 
 def test_prints_a_part_filled_grid_correctly(part_filled_board_output, runner):
     part_filled_board_state = ['x', 'o', '-', '-', '-', '-', '-', '-', '-']
+    game = MockGame(part_filled_board_state)
 
-    assert runner.render_board(part_filled_board_state) == part_filled_board_output
+    assert runner.render_board(game) == part_filled_board_output
 
 
 def test_prints_a_fully_filled_grid(filled_board_output, runner):
     filled_board_state = ['x', 'o', 'x', 'o', 'x', 'o', 'x', 'o', 'x']
+    game = MockGame(filled_board_state)
+
+    assert runner.render_board(game) == filled_board_output
 
-    assert runner.render_board(filled_board_state) == filled_board_output
 
+def test_returns_valid_user_input():
+    mock_io = MockConsoleIO(inputs=['valid'])
+    console = Console(mock_io)
 
-def test_accepts_an_integer_from_the_user():
-    console = Console(MockConsoleIOValidInput())
+    output = console.get_valid_input(['valid'], 'error message')
+    assert output == 'valid'
 
-    assert console.get_int() == 1
 
+def test_continues_to_prompt_for_input_until_valid_input_provided():
+    mock_io = MockConsoleIO(inputs=['invalid', 'valid'])
+    console = Console(mock_io)
 
-def test_errors_if_provided_non_numeric_input(console_with_invalid_io_input):
-    with pytest.raises(Exception) as err:
-        console_with_invalid_io_input.get_int()
+    output = console.get_valid_input(['valid'], 'error message')
 
-    assert "Input was not a number!" in str(err.value)
+    assert mock_io.get_input_call_count == 2
+    assert output == 'valid'
 
 
-def test_sends_message_to_console_io(console_with_valid_io_input):
-    console_io = MockConsoleIOValidInput()
+def prints_an_error_message_if_provided_invalid_input():
```

Comment:
I like the name of the test, it's very clear. Without the name, I'd have struggled to understand what this test is doing just by looking at the code.